tests/test_jenkins.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Santiago Dueñas <sduenas@bitergia.com>
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import argparse
+import json
+import shutil
+import sys
+import tempfile
+import unittest
+
+import httpretty
+
+if not '..' in sys.path:
+    sys.path.insert(0, '..')
+
+from perceval.cache import Cache
+from perceval.errors import CacheError
+from perceval.backends.jenkins import Jenkins, JenkinsCommand, JenkinsClient
+
+JENKINS_SERVER_URL = 'http://example.com/ci'
+JENKINS_JOBS_URL = JENKINS_SERVER_URL+'/view/All/api/json'
+JENKINS_JOB_BUILDS_URL = JENKINS_SERVER_URL + '/job/job1/api/json?depth=2'
+JENKINS_JOB_BUILDS_URL_1 = JENKINS_SERVER_URL + '/job/apex-build-brahmaputra/api/json?depth=2'
+JENKINS_JOB_BUILDS_URL_2 = JENKINS_SERVER_URL + '/job/apex-build-master/api/json?depth=2'
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
+class TestJenkinsBackend(unittest.TestCase):
```

Comment:
You also have to include some tests to check that metadata parameters are right. Specifically, `uuid` and `updated_on`. There are some tests in other backends that check those parameters.